src/tools/clippy/clippy_dev/src/ra_setup.rs

```diff
@@ -51,12 +51,11 @@ fn inject_deps_into_manifest(
 ) -> std::io::Result<()> {
     let extern_crates = lib_rs
         .lines()
-        // get the deps
-        .filter(|line| line.starts_with("extern crate"))
-        // we have something like "extern crate foo;", we only care about the "foo"
-        //              ↓          ↓
-        // extern crate rustc_middle;
-        .map(|s| &s[13..(s.len() - 1)]);
+        // Get the deps
+        //
+        // We have something like "extern crate foo;", we only care about the "foo".
+        // Also we don't trim the string cause we believe rustfmt did that for us.
+        .filter_map(|line| line.strip_prefix("extern crate ").map(|s| s.strip_suffix(';')));
```

Comment:
This is technically a behavior change -- previously this would work if, say, the character after extern crate is a tab. Regardless I'd personally prefer that the clippy changes go into a PR against the clippy repository rather than here, we should really only be landing changes directly to the subtree if they're "required" (i.e., small patches) to avoid merge conflicts as much as possible.